mathics/builtin/arithmetic.py

```diff
@@ -133,7 +134,7 @@ def apply(self, z, evaluation):
             prec = min_prec(*args)
             d = dps(prec)
             args = [
-                Expression(SymbolN, arg, Integer(d)).evaluate(evaluation)
+                _numeric_evaluation_with_prec(arg, evaluation, Integer(d))
```

Comment:
@mmatera @TiagoCavalcante how about if we start calling methods like `apply_` followed by the builtin name.  So here it would be `apply_N`.    Here things are simple in that there is only one `apply` method called `apply`. If the method were called `apply_with_prec` I am not sure what to do, but one simple thing is keep the longer name, even if there is only one of these.

Also, I'd like to start separating such functions like `apply_N` (or  `_numeric_evaluation_with_prec` in a separate module that we can mark for cythonization. 

Your thoughts?

Reply:
I like `apply_N`, it is very easier to understand for everyone who knows what `N` does (and I think that everyone that is going to contribute to Mathics will know it).

About spliting in a module, I like this, also it would be easier to know what needs type hints.

Reply:
OK, then: let's change the name

Reply:
Ok - I'll try to make a stab at this and moving the function into another module later today after I figure out what's up with benchmarking.

Reply:
OK, I will change the name to `apply_N`  later today

Reply:
In any case, this I think should stay in `mathics.builtin.numeric`